src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -72,10 +72,14 @@ public Dictionary(int capacity, IEqualityComparer<TKey> comparer)
         {
             if (capacity < 0) ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
             if (capacity > 0) Initialize(capacity);
-            this.comparer = comparer ?? EqualityComparer<TKey>.Default;
+            if (comparer != EqualityComparer<TKey>.Default)
+            {
+                this.comparer = comparer;
+            }
 
-            if (this.comparer == EqualityComparer<string>.Default)
+            if (typeof(TKey) == typeof(string) && comparer == null)
             {
+                // To start, move off default comparer for string which is randomised
```

Comment:
Are we still going to take this path when somebody passes in `EqualityComparer<string>.Default` explicitly?

Reply:
Yes as earlier test `comparer != EqualityComparer<TKey>.Default` will mean its `null`

Reply:
`this.comparer` will be null. `comparer` is going to be non-null.

Reply:
ah, yes... would like to rename all the class levels - is a bit confusing.

Also, should be done in isolation as it makes any code change confusing.

Reply:
Right

Reply:
fixed